.github/workflows/nightly-uplift.yml

```diff
@@ -0,0 +1,62 @@
+# This workflow automates creation of uplift pull requests.
+# Uplift PR is created daily to uplift the submodule to the latest version.
+
+name: Nighty Uplift PR
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:    
+  uplift-pr:    
+    runs-on: ubuntu-latest    
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0 # Fetch all history and tags
+
+      - name: Set env variable
+        run: |
+          echo "TODAY=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+      - name: Create uplift PR
+        env:
+          GH_TOKEN: ${{ github.token }}
+          RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+          UPLIFT_BRANCH: uplift-${{ env.TODAY }}
+          SUBMODULE_PATH: third_party/tt-mlir
+          SUBMODULE_VERSION: origin/main
+        run: |
+            git config --global user.email ci@tenstorrent.com
+            git config --global user.name "Tenstorrent CI"
+            
+            if git branch -a | grep -q "remotes/origin/$UPLIFT_BRANCH"; then
+              echo "### Branch already exists on remote ###"
+              exit 0
+            fi
+
+            git checkout main
+            git checkout -b $UPLIFT_BRANCH
+            cd $SUBMODULE_PATH
+            git fetch origin
+            git checkout $SUBMODULE_VERSION
+            echo "### Latest commit in $SUBMODULE_PATH ###"
+            git log -1
+            cd ../..
+
+            if [ -z "$(git status --porcelain)" ]; then
+              echo "### No changes in $SUBMODULE_PATH ###"
+              exit 0
+            fi
+            echo "### Commit change and create PR ###"
+            git add $SUBMODULE_PATH
+            git commit -m "Uplift $SUBMODULE_PATH to $SUBMODULE_VERSION $TODAY"
+            git push origin $UPLIFT_BRANCH --force
+
+            gh pr create \
+            --title "Uplift $SUBMODULE_PATH to $SUBMODULE_VERSION $TODAY" \
+            --body "This PR uplifts the $SUBMODULE_PATH submodule to the $SUBMODULE_VERSION." \
+            --base main \
+            --head $UPLIFT_BRANCH \
+            --label uplift \
+            --reviewer ${{ vars.INTEGRATION_OWNER }}
```

Comment:
Where are we specifying the integration owner?

Reply:
@vmilosevic also, can you check out if we can filter notifications for these PRs from our Slack channel? 

@staylorTT was working a bit with these configurations, so he might be able to help you out as well :))

Reply:
> Where are we specifying the integration owner?

This repository variable, we can set this in repo settings->/ variables without code change.

>  this one will create PRs every day, no matter the previous integration

I think it is ok to have fresh check every day, even if there are errors, and this should be triaged every day, maybe it detect more new errors.

As for adjustment option, there is one easy change, we can adjust first condition in script

if git branch -a | grep -q "remotes/origin/$UPLIFT_BRANCH"; then
if git branch -a | grep -q "remotes/origin/uplift-"; then 

to skip creating PRs if there is an active uplift branch. This way it won't create a new branch/PR until the previous one is closed and deleted.

Reply:
Nice, let's do that! That way we'll reduce amount of PRs, and once things are more stable we can have daily uplifts :))